### PR TITLE
builtins: implement ST_MemCollect and ST_MemUnion

### LINKED ISSUE #48984: geo/geomfn: implement ST_MemCollect({geometry})
Implement `ST_MemCollect` on arguments {geometry}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_MemCollect.html).

*Observers: Please react to this issue if you need this functionality.*

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

The following additional guidance has been issued on implementing this function:

Same as ST_Collect.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-08-30T17:39:33Z. Changes to titles, body and labels may be overwritten.</sub>

### LINKED ISSUE #48986: geo/geomfn: implement ST_MemUnion({geometry})
Implement `ST_MemUnion` on arguments {geometry}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_MemUnion.html).

*Observers: Please react to this issue if you need this functionality.*

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

The following additional guidance has been issued on implementing this function:

Same as ST_Union.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-08-30T17:39:33Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by erikgrinaker**
These are implemented as aliases for ST_Collect and ST_Union. In PostGIS
these are memory-optimized versions, but for now it should be sufficient
to simply make them functional.

Release justification: low risk, high benefit changes to existing functionality

Release note (sql change): Implement the geometry aggregate builtins
`ST_MemCollect` and `ST_MemUnion`.

Closes #48984.
Closes #48986.

### CODE REVIEW

**Comment by erikgrinaker** [comment]
CC @otan

File: pkg/sql/sem/builtins/aggregate_builtins.go [hunk 1]
```diff
@@ -400,42 +400,10 @@ var aggregates = map[string]builtinDefinition{
...
-			infoBuilder{
-				info: "Collects geometries into a GeometryCollection or multi-type as appropriate.",
-			}.String(),
-			tree.VolatilityImmutable,
-		),
-	),
+	"st_union":      makeSTUnionBuiltin(),
+	"st_memunion":   makeSTUnionBuiltin(),
+	"st_collect":    makeSTCollectBuiltin(),
+	"st_memcollect": makeSTCollectBuiltin(),
```
Comment (otan) [inline pkg/sql/sem/builtins/aggregate_builtins.go:406]: you can re-use the alias stuff for these: https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go#L4622

also https://github.com/cockroachdb/cockroach/pull/53556 is in the queue and you'll probably have to rebase over the top of them.

thanks!

Comment (otan) [reply pkg/sql/sem/builtins/aggregate_builtins.go:406]: oh nvm you can't :P

Comment (erikgrinaker) [reply pkg/sql/sem/builtins/aggregate_builtins.go:406]: I'm off to bed shortly, so feel free to rebase, otherwise I'll get to it tomorrow.

**Review by otan — APPROVED** [review]

**Pull request merged**